Tsunghao-C/ft_Transcendence
# Game server alpha

Working branch from : https://github.com/Tsunghao-C/ft_Transcendence/pull/38

Added consumers in game_service to handle communications between clients and server related to the game.
Added game.html in game_service/templates to test the game.

Currently only the private match feature is implemented.
A bit laggy, but I won't optimize now.

To test:
Open two clients, each connected to https://localhost:8443/api/game/room/
One client creates a match, the other joins the match, the client creating the match gives the room UID to the client trying to join. Then both clients press ready up and the game begins.

The cleanup logics are not really tested aside from the fact that the game runs, I tried beefing it up but could not get them to work properly so that will be a fight for another day.

Make sure to report game breaking bug I did not foresee, like gameRoom crashes or server sending an error unexpectedly.
